docs/mobile-apps/automated-testing/espresso-xcuitest/xcuitest-introduction.md

```diff
@@ -150,8 +159,87 @@ The next step is to trigger the test by following command:
 ```
 $ saucectl run
 ```
+
 `saucectl` uploads `DemoApp.ipa` and `DemoAppUITests-Runner.ipa` to Sauce VM and executes XCUITest. You can review test results on Sauce UI.
 
+#### Sharding XCUITest
+
+`saucectl` supports executing tests in parallel. To achieve it, `saucectl` needs sharding settings like follows. For more details, check in [yaml configuration](./xcuitest.md#shard).
+
+```
+suites:
+  - name: "sharded suites"
+    shard: concurrency
+    testListFile: test_classes.txt
+```
+
+The `test_classes.txt` can be generated by following steps.
+
+1. Open the project in XCode and select test target
+2. Clik on the `Build Phases`
+3. In the `Build Phases`, click on the `+`-sign
+4. Add the following script
+
+```
+# Set the path to the output file
+OUTPUT_FILE="${SRCROOT}/test_classes.txt"
+
+# Find all Swift files in the test target and exclude the ones you don't want
+TEST_FILES=$(find "${SRCROOT}/${TARGET_NAME}" -name "*.swift" -not -name "Info.plist")
+
+# Print the found test files
+echo "Found test files:"
+echo "$TEST_FILES"
+echo ""
+
+# Initialize an empty string to store class names
+CLASS_NAMES=""
+
+# Get the class names from the Swift files that contain test methods
+while IFS= read -r file; do
+    # Check if the file contains a test method
+    if grep -q "func test[A-Za-z0-9_]*()" "$file"; then
+        # Extract the class name and append it to the list
+        CLASS_NAME=$(grep -Eo "class [a-zA-Z_][a-zA-Z0-9_]*:*" "$file" | awk '{gsub(":", "", $2); print $2}')
+        CLASS_NAMES+="$CLASS_NAME"$'\n'
+    fi
+done <<< "$TEST_FILES"
+
+# Remove duplicate class names
+UNIQUE_CLASS_NAMES=$(echo "$CLASS_NAMES" | sort | uniq | grep -v '^$')
+
+# Write the unique class names to the output file
+echo "$UNIQUE_CLASS_NAMES" > "$OUTPUT_FILE"
+
+# Print the class names and the path to the output file
+echo "Test class names:"
+echo "$UNIQUE_CLASS_NAMES"
+echo ""
+echo "Test class list has been created at: ${OUTPUT_FILE}"
+```
+For the steps also see the below image
+![add-build-script.png](/img/xcuitest/add-build-script.png)
```

Comment:
This screenshot may be a bit hard to follow. Can you send me the image without the callouts? I think I could help with this.
```suggestion
![add-build-script.png](/img/xcuitest/add-build-script.png)
```

Reply:
This image is created by @wswebcreation so I don't have one without callouts. But I can help to take some screenshots from my side.

Reply:
thank you!

Reply:
Just replaced these images. Can you help on that? Thanks!

Reply:
Thank you!